src/sparseml/optim/manager.py

```diff
@@ -57,44 +53,87 @@ def __del__(self):
     def __str__(self) -> str:
         return "\n".join(self.to_string_lines())
 
+    def __eq__(self, compare: object) -> bool:
+        return str(self) == str(compare)
```

Comment:
I want to keep this so that it can generally compare to raw strings and yaml objects as well to say that the a manager is equivalent to a yaml file on disk or in memory. Given how complex the yamls are, we don't risk an improper collision

Reply:
👍

Reply:
in that vein, nice to have would be serializing and then deserializing string inputs to Manager objects and back again. that will standardize stuff like different yaml formats, ordering, whitespace, etc